packages/docusaurus-logger/src/index.ts

```diff
@@ -0,0 +1,134 @@
+  pico.blue(pico.bold(msg));
+const code: import('picocolors/types').Formatter = (msg) =>
+  pico.cyan(`\`${msg}\``);
+const subdue = pico.gray;
+const num = pico.yellow;
+
+function interpolate(
+  msgs: TemplateStringsArray,
+  ...values: InterpolatableValue[]
+): string {
+  let res = '';
+  values.forEach((value, idx) => {
+    const flag = msgs[idx].match(/%[a-z]+$/);
+    res += msgs[idx].replace(/%[a-z]+$/, '');
+    const format = (function () {
+      if (!flag) {
+        return (a: string | number) => a;
+      }
+      switch (flag[0]) {
+        case '%p':
+          return path;
+        case '%n':
+          return num;
+        case '%i':
+          return id;
+        case '%s':
+          return subdue;
+        case '%c':
+          return code;
+        default:
+          throw new Error(
+            'Bad Docusaurus logging message. This is likely an internal bug, please report it.',
+          );
+      }
+    })();
+    res += Array.isArray(value)
+      ? `\n- ${value.map((v) => format(v)).join('\n- ')}`
+      : format(value);
+  });
+  res += msgs.slice(-1)[0];
+  return res;
+}
+
+function info(msg: any): void;
+function info(
+  msg: TemplateStringsArray,
+  ...values: [InterpolatableValue, ...InterpolatableValue[]]
+): void;
+function info(msg: any, ...values: InterpolatableValue[]): void {
+  console.info(
+    `${pico.cyan(pico.bold('[INFO]'))} ${
+      values.length === 0
+        ? msg
+        : interpolate(msg as TemplateStringsArray, ...values)
+    }`,
+  );
+}
+function warn(msg: any): void;
+function warn(
+  msg: TemplateStringsArray,
+  ...values: [InterpolatableValue, ...InterpolatableValue[]]
+): void;
+function warn(msg: any, ...values: InterpolatableValue[]): void {
+  console.warn(
+    pico.yellow(
+      `${pico.bold('[WARNING]')} ${
+        values.length === 0
+          ? msg
+          : interpolate(msg as TemplateStringsArray, ...values)
+      }`,
+    ),
+  );
+}
+function error(msg: any): void;
+function error(
+  msg: TemplateStringsArray,
+  ...values: [InterpolatableValue, ...InterpolatableValue[]]
+): void;
+function error(msg: any, ...values: InterpolatableValue[]): void {
+  console.error(
+    pico.red(
+      `${pico.bold('[ERROR]')} ${
+        values.length === 0
+          ? msg
+          : interpolate(msg as TemplateStringsArray, ...values)
+      }`,
+    ),
+  );
+}
+function success(msg: any): void;
+function success(
+  msg: TemplateStringsArray,
+  ...values: [InterpolatableValue, ...InterpolatableValue[]]
+): void;
+function success(msg: any, ...values: InterpolatableValue[]): void {
+  console.log(
+    `${pico.green(pico.bold('[SUCCESS]'))} ${
+      values.length === 0
+        ? msg
+        : interpolate(msg as TemplateStringsArray, ...values)
+    }`,
+  );
+}
+
+const logger = {
+  ...pico,
+  path,
+  id,
+  code,
+  subdue,
+  num,
+  interpolate,
+  info,
+  warn,
+  error,
+  success,
+};
+
+export default logger;
```

Comment:
Considering we'll want to provide configurable logging levels later, maybe exposing a static logger instance like this is not ideal? Not sure yet what our final API will look like though, we'll probably need more reactors later anyway

Reply:
`logger` behaves like `console` in every way. I plan to make it read env vars when we support logging levels.

Reply:
Later I'd like all loggers to have a name, they should rather be created and added to some context, eventually log their name to the console. We'd call something like `createLogger()`

Config could look like this:


```js
const siteConfig = {
  logging: {
    core: "warn"
    browser: "error",
    plugins: {
      "docusaurus/plugin-content-docs": "info",
      "docusaurus/plugin-content-blog": "info",
    }
  }
}
```

We can see how to implement that later though